ukvm/ukvm_module_net.c

```diff
@@ -178,8 +178,10 @@ static void hypercall_netwrite(struct ukvm_hv *hv, ukvm_gpa_t gpa)
     int ret;
 
     ret = write(netfd, UKVM_CHECKED_GPA_P(hv, wr->data, wr->len), wr->len);
-    assert(wr->len == ret);
-    wr->ret = 0;
+    if (wr->len == ret)
+        wr->ret = 0;
+    else
+        wr->ret = -1;
 }
```

Comment:
That was actually the issue I observed. I started the unikernel without setting the interface up. With the change this sequence of events is not fatal any longer:

- Start test_ping_server unikernel
- Bring up and configure tap interface
- Ping 10.0.0.2

Looking at the error handling code in [test_ping_serve.c ](https://github.com/Solo5/solo5/blob/master/tests/test_ping_serve/test_ping_serve.c#L248) which checks if net write returns -1 (cannot happen with the current code) gave me the impression that it was intended for users to be able to handle errors.